internal/ui/ui_glfw.go

```diff
@@ -200,38 +196,101 @@ func initialize() error {
 	return nil
 }
 
-type monitor struct {
+func setInitMonitor(m *glfw.Monitor) {
+	theUI.initMonitor = m
+	theUI.initDeviceScaleFactor = theUI.deviceScaleFactor(m)
+	// GetVideoMode must be called from the main thread, then call this here and record
+	// initFullscreen{Width,Height}InDIP.
+	v := m.GetVideoMode()
+	theUI.initFullscreenWidthInDIP = int(theUI.dipFromGLFWMonitorPixel(float64(v.Width), m))
+	theUI.initFullscreenHeightInDIP = int(theUI.dipFromGLFWMonitorPixel(float64(v.Height), m))
+}
+
+// Monitor is a wrapper around glfw.Monitor.
+type Monitor struct {
 	m  *glfw.Monitor
 	vm *glfw.VidMode
 	// Pos of monitor in virtual coords
-	x int
-	y int
+	x      int
+	y      int
+	width  int
+	height int
+	id     int
+	name   string
+}
+
+// Bounds returns the monitor's bounds.
+func (m *Monitor) Bounds() image.Rectangle {
+	return image.Rect(m.x, m.y, m.x+m.width, m.y+m.height)
+}
+
+// Name returns the monitor's name.
+func (m *Monitor) Name() string {
+	return m.name
 }
 
 // monitors is the monitor list cache for desktop glfw compile targets.
 // populated by 'updateMonitors' which is called on init and every
 // monitor config change event.
 //
 // monitors must be manipulated on the main thread.
-var monitors []*monitor
+var monitors []*Monitor
+
+// AppendMonitors appends the current monitors to the passed in mons slice and returns it.
+func (u *userInterfaceImpl) AppendMonitors(mons []*Monitor) []*Monitor {
+	u.m.RLock()
+	defer u.m.RUnlock()
+	return append(mons, monitors...)
+}
+
+// Monitor returns the window's current monitor. Returns nil if there is no current monitor yet.
+func (u *userInterfaceImpl) Monitor() *Monitor {
+	u.m.RLock()
+	defer u.m.RUnlock()
+	var glfwMonitor *glfw.Monitor
+	if u.isFullscreen() {
+		glfwMonitor = u.window.GetMonitor()
+	} else {
+		glfwMonitor = u.currentMonitor()
+	}
+	if glfwMonitor == nil {
+		return nil
+	}
+	for _, m := range monitors {
+		if m.m.IsMonitor(glfwMonitor) {
```

Comment:
Do we need a general `IsMonitor`? What about using indices/ids to compare monitors?

Reply:
There are no indices/IDs to compare. This is due to:

```go
func (w *Window) GetMonitor() *Monitor {
	m := w.w.GetMonitor()
	if m == nil {
		return nil
	}
	return &Monitor{m}
}
```

On Windows we have (glfw/glfw_windows.go:115):

```go
func (w *Window) GetMonitor() *Monitor {
	m, err := (*goglfw.Window)(w).GetMonitor()
	if err != nil {
		panic(err)
	}
	return (*Monitor)(m)
}
```

However, see latest commit for an alternative approach... Removing the call to `GetMonitor()` in `monitorFromWindow(...)` fixes the issue. I do not know if removing this is a good idea, as I do not know why it was there in the first place.